### ISSUE TITLE
can a template action be based on the "current selection" in the editor?

### ISSUE THREAD

**Issue by luckman212** [SilentVoid13/Templater#44]
Not sure if I am even asking this in the right way, but here's my use case:

I'd like to create a template that uses the current selected word or phrase as a parameter for the template. One example could be a "turn selection into linked Google search":

1. hightlight a word e.g. _"Zettel"_
2. activate Templater, with template containing something like:
```
[{{tp_selection}}](https://www.google.com/search?q={{tp_selection:urlencoded=true}})
```
3. have that selection replaced with `[Zettel](https://www.google.com/search?q=Zettel)`

Is this possible today or somehow in the future?

**Comment by luckman212** [comment]
current hack is to copy the word to the clipboard first and then use this as the template src:

```bash
w=$(pbpaste)
q=$(python3 -c 'import sys; from urllib.parse import quote; print(quote(sys.argv[1]));' "$w")
echo "[$w](https://www.google.com/search?q=$q)"
```

**Comment by luckman212** [comment]
Seems operating on the current selection is indeed possible, see https://github.com/lukeleppan/better-word-count/blob/master/src/main.ts#L33 for an example

**Comment by SilentVoid13** [comment]
Hello @luckman212, that's a good internal template idea !
I'll probably add this one in the next release, thanks for the suggestion.

**Comment by SilentVoid13** [comment]
Hey @luckman212, sorry for the late response, this has been implemented as of Templater 1.0.0!
If you update, you will now be able to use `<% tp.file.selection() %>` to access your current selection content.
Tell me if this works for you.
Closing this.
